homeassistant/components/control4/director_utils.py

```diff
@@ -39,24 +29,5 @@ async def director_update_data(
 
 
 async def refresh_tokens(hass: HomeAssistant, entry: ConfigEntry):
-    """Store updated authentication and director tokens in hass.data."""
-    config = entry.data
-    account_session = aiohttp_client.async_get_clientsession(hass)
-
-    account = C4Account(config[CONF_USERNAME], config[CONF_PASSWORD], account_session)
-    await account.getAccountBearerToken()
-
-    controller_unique_id = config[CONF_CONTROLLER_UNIQUE_ID]
-    director_token_dict = await account.getDirectorBearerToken(controller_unique_id)
-    director_session = aiohttp_client.async_get_clientsession(hass, verify_ssl=False)
-
-    director = C4Director(
-        config[CONF_HOST], director_token_dict[CONF_TOKEN], director_session
-    )
-    director_token_expiry = director_token_dict["token_expiration"]
-
-    _LOGGER.debug("Saving new tokens in hass data")
-    entry_data = hass.data[DOMAIN][entry.entry_id]
-    entry_data[CONF_ACCOUNT] = account
-    entry_data[CONF_DIRECTOR] = director
-    entry_data[CONF_DIRECTOR_TOKEN_EXPIRATION] = director_token_expiry
+    """Rerun setup to update tokens for all entities."""
+    await hass.config_entries.async_reload(entry.entry_id)
```

Comment:
This seems like a really overkill solution, it will make all devices unavailable and look unstable in between this happening.

We should look into fixing the passing of the new token to the entities instead.